src/freenas/usr/local/bin/truenas-grub.py

```diff
@@ -15,6 +16,13 @@ def get_serial_ports():
     advanced = query_config_table("system_advanced", prefix="adv_")
     kernel_extra_options = advanced.get("kernel_extra_options") or ""
 
+    # check for /data/.vendor
+    try:
+        with open("/data/.vendor", "r") as f:
+            vendor = json.loads(f.read()).get("name", "TrueNAS Scale")
+    except FileNotFoundError:
+        vendor = "TrueNAS Scale"
+
```

Comment:
Using the vendor plugin this becomes
```
with Client() as c:
    vendor = c.call("system.vendor.name") or "TrueNAS Scale"
```
Not sure if this is the preferred way to do it.

Reply:
I think that since that would require extra imports, it'd be safe to keep it the way it is reading the JSON. We also run truenas-grub from scale-build, I'm not entirely sure what the context looks like there, so just to be safe there too.

Reply:
We can't use middleware client here, middleware is not running during ISO install